src/controller/python/chip/clusters/ClusterObjects.py

```diff
@@ -369,6 +370,19 @@ def _cluster_object(cls) -> ClusterObject:
 
 
 class ClusterEvent(ClusterObject):
+    def __init_subclass__(cls, *args, **kwargs) -> None:
+        """Register a subclass."""
+        super().__init_subclass__(*args, **kwargs)
+        try:
+            if cls.cluster_id not in ALL_EVENTS:
```

Comment:
if we make `ALL_EVENTS = collections.defaultdict(dict)` we could always just directly do `ALL_EVENTS[cls.cluster_id][cls.event_id] = cls`.

Or do we want to be able to get the KeyNotFound error instead of a default empty dict?

Reply:
Would that return an empty dict on access too? I don't want any misleading returns here - if the thing doesn't exist and you try to access it, it should fail loudly.

But I honestly just did it this way because it matches the ALL_ATTRIBUTES above.

Reply:
I'll chime in here as I added the ALL_ATTRIBUTES and ALL_CLUSTERS but imo its more pythonic to just raise on the KeyError than secretly hide things (in this case with a default dict). Not sure if a key lookup still works (so fails if the key doesnt exist) with a defaultdict in place for example.